rust/stackable-cockpit/src/xfer/cache.rs

```diff
@@ -128,34 +128,34 @@ impl Cache {
 
     /// Removes all cached files by deleting the base cache folder and then
     /// recreating it.
-    pub async fn purge(&self, only_old_files: bool) -> Result<()> {
+    pub async fn purge(&self, delete_filter: DeleteFilter) -> Result<()> {
         match &self.backend {
             CacheBackend::Disk { base_path } => {
                 let mut entries = fs::read_dir(base_path).await.context(CacheIoSnafu)?;
 
                 while let Some(entry) = entries.next_entry().await.context(CacheIoSnafu)? {
                     let metadata = entry.metadata().await.context(CacheIoSnafu)?;
 
-                    // Skip the last-auto-purge file
-                    if metadata.is_file() && is_protected_file(entry.file_name())? {
-                        continue;
-                    }
-
-                    // With --old / --outdated
-                    if only_old_files
-                        && metadata
-                            .modified()
-                            .context(CacheIoSnafu)?
-                            .elapsed()
-                            .context(SystemTimeSnafu)?
-                            > self.max_age
-                    {
+                    let should_delete_file = match delete_filter {
+                        // Skip the last-auto-purge file
```

Comment:
```suggestion
                        // Skip protected files
```
Might as well align with L115.

Reply:
Already done, heh.

Reply:
Done in 2eaaa8b